sidestream-tech/unified-auctions-ui
# Added `Notification-Service` for contract events

closes #235 

For the first version I made the following simple UI:

Console view:
![Screenshot 2022-05-04 at 08 06 31](https://user-images.githubusercontent.com/30908158/166630481-c0087724-95cc-46a3-830c-896a48a68d9f.png)

Email Template:
![Screenshot 2022-05-04 at 08 31 15](https://user-images.githubusercontent.com/30908158/166632473-74f77011-731e-4a07-979a-5ee0b8725b53.png)
_Email links to [Etherscan](https://kovan.etherscan.io/tx/0xb7d44d618366f5a6158c4889edfd13a62644486da8f87ed0e6af3b0b2d4938f7#eventlog)._

I was thinking that this would be enough as a first version, as you can view the data on etherscan. I can work on a second PR afterwards that takes care of actually displaying the differences in the email. In that PR I will also include the functionality to customize the email message per subscription, as outlined by the issue. However I wanted to get a first version up and running before diving into this. I hope this is okay!

## How to test @LukSteib 

As the event we later want to observe, does not have many updates, I added a second, quite frequently used event as a test. 

In `notifcation-service/src/constants/SUBSCRIPTIONS.ts` You can re-add the following lines:

https://github.com/sidestream-tech/unified-auctions-ui/blob/add-notification-service/notification-service/src/constants/SUBSCRIPTIONS.ts#L24-L28

_Note: The commented out contract only works on the  `kovan` network, so be sure to test it on there or use a alternative contract._

### Email setup
By default, if no SMTP values are provided, I connect with a service called [Ethereal Email](https://ethereal.email/), as this was included in my library. However, this service is down today 🙃, therefore you need to add your own `SMTP` settings.

I would advise to use a service called [MailTrap](https://mailtrap.io/). Once you logged in and click on your inbox you will find a dropdown titled `Integrations`. By default it should have something called `cURL` selected. Select `NodeMailer` from the list. 

You can then set the environment variables to the following:
- SMTP_HOST="smtp.mailtrap.io"
- SMTP_PORT="2525"
- SMTP_USERNAME="YOUR_USER_ID"
- SMTP_PASSWORD="YOUR_USER_PASSWORD"

What is also a bit different in this project is how to start it. Running `npm run dev` will only live compile it, but not execute it. Therefore you should run `npm run start`. This will automatically build and start the service.
